x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -148,5 +185,16 @@ private void assertRecovered(String indexName, TotalHits originalAllHits, TotalH
                     lessThanOrEqualTo(1)); // we make a new commit so we write a new `segments_n` file
             }
         }
+
+        for (int i = 0; i < threads.length; i++) {
+            threads[i].join();
+
+            final TotalHits allTotalHits = allHits.get(i);
+            final TotalHits barTotalHits = barHits.get(i);
+
+            logger.info("--> thread #{} has [{}] hits in total, of which [{}] match the query", i, allTotalHits, barTotalHits);
+            assertThat(allTotalHits, equalTo(originalAllHits));
+            assertThat(barTotalHits, equalTo(originalBarHits));
```

Comment:
I got a test failure here, but there was nothing to suggest why in the log:

```
java.lang.AssertionError: 
Expected: <4003 hits>
     but: was <3564 hits>
Expected :<4003 hits>
Actual   :<3564 hits>
 <Click to see difference>


	at __randomizedtesting.SeedInfo.seed([D2D6A1BF6B71726C:58B5D6C4F8E623DF]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.assertRecovered(SearchableSnapshotsIntegTests.java:197)
	at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testCreateAndRestoreSearchableSnapshot(SearchableSnapshotsIntegTests.java:134)
```

Reply:
Is it a different failure from https://github.com/elastic/elasticsearch/pull/50693#discussion_r368672278 or a consequence of it?

Reply:
They weren't obviously related. I found this one when re-running the tests in a loop to try and reproduce the one you linked.